### PR TITLE
Milestone 2: Components

### PR DESCRIPTION
**Pull request by HeflerDev**
Requirements:

- [x] Prepare the directory structure
  - [x] Create a components directory in the src folder.
  - [x] Create four files in there, each one representing one React component. For instance, the App component should be placed in src/components/App.js.
- [x] Implement the App component
- [x] App should render the Display and ButtonPanel component. HINT: at this moment App component is a stateless component as it is not fully implemented yet. In order to avoid linter errors remember to make it a functional component..
- [x] Implement the Display component
  - [x] It should accept the result of the calculation as an incoming prop.
  - [x] The result prop should be a String.
  - [x] The default value of the result prop should be 0.
- [x] Implement the Button component
  - [x] It should accept a button name as a prop.
  - [x] The button name prop should be a String.
- [x] Implement the ButtonPanel component
  - [x] It should render the calculator panel with all the buttons in the following groups: (use divs to divide buttons into groups)
  - [x] Group 1: AC, +/-, %, ÷
  - [x] Group 2: 7, 8, 9, X
  - [x] Group 3: 4, 5, 6, -
  - [x] Group 4: 1, 2, 3, +
  - [x] Group 5: 0, ., =

### CODE REVIEW

**Comment by Torres-ssf** [comment]
**Invalid Submission**
Hello @HeflerDev :hand: 
I can't review your project right now because it's considered an invalid submission due to the following reasons:
- The linter was not setup in the repository.
- A descriptive README file was not implemented.

File: README.md [hunk 1]
```diff
@@ -1,70 +1,33 @@
-# Getting Started with Create React App
+# React Calculator - By HeflerDev
```
Comment (bellom) [inline README.md:1]: ### [Optional]
Do well to add more information to your readme file, take a look at the [Microverse template](https://github.com/microverseinc/readme-template) from pathwright

File: src/components/ButtonPanel.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      <div>3</div>
+      <div>+</div>
+    </div>
+    <div>
+      <div>0</div>
+      <div>.</div>
+      <div>=</div>
+    </div>
+  </>
+);
```
Comment (bellom) [inline src/components/ButtonPanel.js:35]: ### [Optional]
Your code appears to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In other words; you repeated the `Button` component multiple times.

You can use [objects.key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)  and `map` to loop through the objects. `objects.keys` for the outer loop and `map` for the inner loop.

Example; 
```
const GroupButtons = {
  Group 1: ['AC', '+/-', '%', '÷'],
  Group 2: ['7', '8', '9', 'X'],
   ............
}
```
OR

[Creating a two-dimensional array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#creating_a_two-dimensional_array) and implement nested loop with a map.
EXample;
```
  const buttons = [
    ['AC', '+/-', '%', '÷'],
    ['7', '8', '9', 'X'],
    ..........
  ];
```

**Review by bellom — APPROVED** [review]
## Hi @HeflerDev :clap: 

##### You have done wonderfully well, you passed all checklist for this milestone :+1:
 
# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck with your next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun J. Bello**
**_Slack Handle - @sean bellom_**

File: src/components/ButtonPanel.js (hunk 2 above)
Comment (HeflerDev) [reply src/components/ButtonPanel.js:35]: Nice! I used Emmet to type for me lol, but I will surely implement your suggestion.
Thanks a lot! :+1:

**Pull request merged**